auth/role_info.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "auth/role_info.hh"
+
+// Seastar features.
+#include <seastar/core/print.hh>
+
+// Scylla includes.
+#include "cql3/util.hh"
+#include "seastar/coroutine/maybe_yield.hh"
+
+namespace auth {
+
+std::ostream& role_info::describe(std::ostream& os) const {
+    // Note: fmt prints boolean values as `true` / `false` if the presentation type is not specified.
+
+    const auto formatted_role_name = cql3::util::maybe_quote(role_name);
+
+    if (maybe_salted_hash) {
+        // Note: Although `crypt` functions don't produce whitespace characters, etc., they don't guarantee
+        //       that the salted hash won't contain a quotation mark:
+        //
+        //         "Hashed passphrases are always entirely printable ASCII, and do not contain any whitespace
+        //          or the characters `:`, `;`, `*`, `!`, or `\`."
+        //
+        //       --- crypt(5); I changed the formatting of the manual page a bit.
+        //
+        //       Because of that, we call `cql3::util::maybe_quote` for safety.
```

Comment:
The comment seems outdated, `maybe_quote` is not used here but rather `single_quote`.